### PR TITLE
Project: Modernize SkyBlock

### PR DESCRIPTION
**Pull request by GiantQuartz**
Objectives of this Pull Request:
- [x] Create a system to support multiple data providers 
- [x] Replace the current way to get user data (directly from the JSON files) with a Session-based system
- [x] Create a custom command map to allow developers to create their own commands
- [x] Improve source readability and start using PHP 7 features
- [x] Implement translations
- [x] Implement island levels
- [ ] Implement all old commands
- [ ] Implement island levels (suggested by @Shelly7w7 )
- [ ] Add events
- [x] Add custom starting contents per generator (suggested by @95CivicSi )

### CODE REVIEW

File: plugin.yml [hunk 1]
```diff
@@ -1,65 +1,7 @@
...
-    description: Delete your island
-  sbpe.cmd.makeleader:
-    default: true
-    description: make someone else the leader of your island
-  sbpe.cmd.version:
-    default: true
-    description: get version of SBPE
-...
+version: 1.0.0
+api: [4.0.0]
```
Comment (iiFlamiinBlaze) [inline plugin.yml]: Should be 3.0.0, 4.0.0 is not released yet and its not even close to release. Do not support 4.0.0!

**Comment by SuperAdam47** [comment]
İ am tested
image

**Comment by GiantQuartz** [comment]
@SuperAdam47 This still under development.

**Comment by GiantQuartz** [comment]
By the way, if anyone has a good idea, I'm all ears. I'd love to know what kind of features you'd like to see in this plugin. 💯 :)

**Comment by Shelly7w7** [comment]
Levels xd?

**Comment by SuperAdam47** [comment]
add VIP and VIP+ ısland generator? :)

**Comment by 95CivicSi** [comment]
@SuperAdam47 What would you expect the differences of those generators to be versus the BasicIsland?

**Comment by Shelly7w7** [comment]
@95civicsi can you add island levels?

**Comment by 95CivicSi** [comment]
@Shelly7w7 What do you mean by "Levels"?

**Comment by Shelly7w7** [comment]
@95civicsi i meant, like when u place ores on your island, your island levels up. e.g grass xp is 90 and whenever u place grass you get xp and your island levels up like pc servers

**Comment by SuperAdam47** [comment]
Could it be a VIP ready house?

**Comment by 95CivicSi** [comment]
The chest inventory isn't custom per generator yet but that's in the works.

**Comment by GiantQuartz** [comment]
Good news! This branch is now functional (with a minimal amount of features).
I'll be implementing new features this week, be ready, a lot of cool things are coming to this SkyBlock plugin!

**Pull request merged**